### PR TITLE
Post storage

### PR DESCRIPTION
**Pull request by leohtkam**
Any posts in progress in a conversation (new post, summary, resolution that user haven't submitted) are now saved as long as user doesn't leave the site.

TODO:
Warn user when they try to leave the site when they have unsubmitted posts in progress.

### CODE REVIEW

**Comment by nickfong** [comment]
You can probably get rid of 70febcd and 7dd3139 since 70febcd corresponds to the old version of the commit that Wenson pushed up earlier (and that has since been merged).  Will CR after finishing fixing seeds, assuming someone else hasn't done so already

File: app/assets/javascripts/homepage/conversation.coffee [hunk 1]
```diff
@@ -235,3 +259,31 @@ conversation.controller("ConversationController", ["$scope", "$stateParams", "AP
...
+        getConversation(conversationId).resolutionText = text
+    this.getAddPostText = (conversationId) ->
+        return getConversation(conversationId).addPostText
+    this.getSummaryText = (conversationId) ->
+        return getConversation(conversationId).summaryText
+    this.getResolutionText = (conversationId) ->
+        return getConversation(conversationId).resolutionText
+    this.hasUnsavedProgress = (conversationId) ->
+        c = getConversation(conversationId)
+        return (c.addPostText || c.summaryText || c.resolutionText)
```
Comment (whsieh) [inline app/assets/javascripts/homepage/conversation.coffee]: Heads up: this returns a string, not a boolean. It probably works still, but I'd rather it be `c.addPostText != "" or .....`

File: app/assets/javascripts/homepage/conversation.coffee [hunk 2]
```diff
@@ -208,6 +230,8 @@ conversation.controller("ConversationController", ["$scope", "$stateParams", "AP
 
     $rootScope.$on "conversationPageWillLoad", (scope, args) ->
         if conversationId isnt args.conversationId then clearInterval conversationPollingProcess
+    $rootScope.$on '$locationChangeStart', (event, next, current) ->
```
Comment (whsieh) [inline app/assets/javascripts/homepage/conversation.coffee]: Double quotes over single quotes, as per style guidelines.

**Comment by whsieh** [comment]
+1 after fixes.

**Comment by nickfong** [comment]
+1 I'll merge in now

**Pull request merged**